staging/src/k8s.io/client-go/dynamic/fake/simple_test.go

```diff
@@ -446,6 +455,396 @@ func TestListWithTypedFixtures(t *testing.T) {
 	}
 }
 
+func TestDelete(t *testing.T) {
+	deleteTests := map[string]struct {
+		ns            string
+		name          string
+		deleteOptions metav1.DeleteOptions
+		subresources  []string
+		wantActions   []clientgotesting.Action
+	}{
+		"resource deletion with namespace": {
+			name: "name-foo",
+			ns:   "ns-foo",
+			deleteOptions: metav1.DeleteOptions{
+				DryRun: []string{metav1.DryRunAll},
+			},
+			wantActions: []clientgotesting.Action{
+				clientgotesting.DeleteActionImpl{
+					Name: "name-foo",
+					ActionImpl: clientgotesting.ActionImpl{
+						Namespace: "ns-foo",
+						Verb:      "delete",
+						Resource: schema.GroupVersionResource{
+							Group: "group", Version: "version", Resource: "thekinds"},
+					},
+					DeleteOptions: metav1.DeleteOptions{
+						DryRun: []string{metav1.DryRunAll},
+					},
+				},
+			},
+		},
+		"resource deletion with no namespace": {
+			name: "name-foo",
+			deleteOptions: metav1.DeleteOptions{
+				DryRun: []string{metav1.DryRunAll},
+			},
+			wantActions: []clientgotesting.Action{
+				clientgotesting.DeleteActionImpl{
+					Name: "name-foo",
+					ActionImpl: clientgotesting.ActionImpl{
+						Verb: "delete",
+						Resource: schema.GroupVersionResource{
+							Group: "group", Version: "version", Resource: "thekinds"},
+					},
+					DeleteOptions: metav1.DeleteOptions{
+						DryRun: []string{metav1.DryRunAll},
+					},
+				},
+			},
+		},
+		"subresource deletion with namespace": {
+			ns:           "ns-foo",
+			name:         "test",
+			subresources: []string{"status"},
+			deleteOptions: metav1.DeleteOptions{
+				OrphanDependents: ptr.To[bool](true),
+			},
+			wantActions: []clientgotesting.Action{
+				clientgotesting.DeleteActionImpl{
+					Name: "test",
+
+					ActionImpl: clientgotesting.ActionImpl{
+						Namespace:   "ns-foo",
+						Subresource: "status",
+						Verb:        "delete",
+						Resource: schema.GroupVersionResource{
+							Group: "group", Version: "version", Resource: "thekinds"},
+					},
+					DeleteOptions: metav1.DeleteOptions{
+						OrphanDependents: ptr.To[bool](true),
+					},
+				},
+			},
+		},
+		"subresource deletion with no namespace": {
+			name:         "test",
+			subresources: []string{"status"},
+			deleteOptions: metav1.DeleteOptions{
+				OrphanDependents: ptr.To[bool](true),
+			},
+			wantActions: []clientgotesting.Action{
+				clientgotesting.DeleteActionImpl{
+					Name: "test",
+					ActionImpl: clientgotesting.ActionImpl{
+						Subresource: "status",
+						Verb:        "delete",
+						Resource: schema.GroupVersionResource{
+							Group: "group", Version: "version", Resource: "thekinds"},
+					},
+					DeleteOptions: metav1.DeleteOptions{
+						OrphanDependents: ptr.To[bool](true),
+					},
+				},
```

Comment:
For completeness can we also add a test case to TestDelete for empty deleteOptions?